plugins/gcov/README.md

```diff
@@ -23,79 +23,239 @@ pip install gcovr
+  # Defaults to disabled. (gcovr --xml-pretty)
+  :xml_pretty: [true|false]
+  :cobertura_pretty: [true|false]
+
+  # Cobertura XML report filename.
+  :xml_artifact_filename: <output>
+  :cobertura_artifact_filename: <output>
 ```
+
+### SonarQube XML Reports
+
+Generation of SonarQube XML reports may be modified with the following configuration items.
+
+```yaml
 :gcov:
-  :html_report_type: basic
+  # SonarQube XML report filename.
+  :sonarqube_artifact_filename: <output>
 ```
 
-To create a detailed HTML report, with line by line breakdown of the
-coverage, use the following config.
+### JSON Reports
 
+Generation of JSON reports may be modified with the following configuration items.
+
+```yaml
+:gcov:
+  # Set to 'true' to pretty-print the JSON report, otherwise set 'false'.
+  # Defaults to disabled. (gcovr --json-pretty)
+  :json_pretty: [true|false]
+
+  # JSON report filename.
+  :json_artifact_filename: <output>
 ```
+
+### Text Reports
+
+Generation of text reports may be modified with the following configuration items.
+Text reports may be printed to the console or output to a file.
+
+```yaml
 :gcov:
-  :html_report_type: detailed
+  # Text report filename.
+  # The text report is printed to the console when no filename is provided.
+  :text_artifact_filename: <output>
 ```
 
+### Common Report Options
+
 There are a number of options to control which files are considered part of
 the coverage report. Most often, we only care about coverage on our source code, and not
-on tests or automatically generated mocks, runners, etc. However, there are times 
-where this isn't true... or there are times where we've moved ceedling's directory 
+on tests or automatically generated mocks, runners, etc. However, there are times
+where this isn't true... or there are times where we've moved ceedling's directory
 structure so that the project file isn't at the root of the project anymore. In these
-cases, you may need to tweak the following:
+cases, you may need to tweak `report_include`, `report_exclude`, and `exclude_directories`.
 
-```
-:gcov:
-  :report_root: "."
-  :report_exclude: "^build|^vendor|^test|^support"
-  :report_include: "^src"
-```
-
-One important note about html_report_root: gcovr will only take a single root folder, unlike 
-Ceedling's ability to take as many as you like. So you will need to choose a folder which is 
+One important note about `report_root`: gcovr will take only a single root folder, unlike
+Ceedling's ability to take as many as you like. So you will need to choose a folder which is
 a superset of ALL the folders you want, and then use the include or exclude options to set up
 patterns of files to pay attention to or ignore. It's not ideal, but it works.
 
-Finally, there are a number of settings which can be specified in order to adjust the
+Finally, there are a number of settings which can be specified to adjust the
 default behaviors of gcov:
```

Comment:
I think these options specify behaviours of gcovr not gcov. For now it is ok, because we only have the gcovr, but after extend for the ReportGenerator it may not be true.

Reply:
Updated `gcov` to `gcovr` as requested.

Reply:
Do we want to put all gcovr specific options under a :gcovr: section?

```yaml
:gcov:
    :gcovr:
        :report_root: "."
        :keep: true
    :report_generator:
        :verbosity: Info
```